pytest_logger/plugin.py

```diff
@@ -97,9 +99,21 @@ def pytest_runtest_teardown(self, item, nextitem):
         if logger:
             logger.on_teardown()
 
+    @pytest.mark.hookwrapper
     def pytest_runtest_makereport(self, item, call):
+        outcome = yield
+        tr = outcome.get_result()
         logger = getattr(item, '_logger', None)
         if logger:
+            if self._logsdir and self._div_by_res_logdir and tr.outcome in self._div_by_res_subdirs:
+                div_by_res_logdir = self._logsdir.join(self._div_by_res_logdir, tr.outcome)
+                div_by_res_logdir.ensure(dir=1)
```

Comment:
ensures the dir prematurely. subdir is going to be ensured in a while. Consider using single `join`.

Reply:
i thought that only the last dir will get "ensured" - it appears that it works with a whole path . cool!